editor/plugins/gizmos/navigation_region_3d_gizmo_plugin.h

```diff
@@ -41,7 +41,13 @@ class NavigationRegion3DGizmoPlugin : public EditorNode3DGizmoPlugin {
 		Vector3 to;
 
 		static uint32_t hash(const _EdgeKey &p_key) {
-			return HashMapHasherDefault::hash(p_key.from) ^ HashMapHasherDefault::hash(p_key.to);
+			uint32_t hash = hash_murmur3_one_real(p_key.from.x);
+			hash = hash_murmur3_one_real(p_key.from.y, hash);
+			hash = hash_murmur3_one_real(p_key.from.z, hash);
+			hash = hash_murmur3_one_real(p_key.to.x, hash);
+			hash = hash_murmur3_one_real(p_key.to.y, hash);
+			hash = hash_murmur3_one_real(p_key.to.z, hash);
+			return hash_fmix32(hash);
```

Comment:
To avoid errors here I'd suggest using the dedicated methods for vectors, and combine them like is done with `PairHash`:
https://github.com/godotengine/godot/blob/03e6fbb010c3546593bd91a0dabc045a9882705a/core/templates/pair.h#L72-L79

Reply:
The problem with vector hashing is that the `hash_fmix32` method is called more than once.
https://github.com/godotengine/godot/blob/03e6fbb010c3546593bd91a0dabc045a9882705a/core/templates/hashfuncs.h#L337
And the `hash_one_uint64` method is unnecessary. This not only reduces performance, but also makes the hash not as good. Since I checked the combination of `hash_one_uint64` and `hash_fmix32`, there are much more collisions.